### PR TITLE
Adds example/example.md for the example page on pub.dev

### PR DESCRIPTION
**Pull request by robert-northmind**
## Description

We didn't have a good example page for pub.dev for WidgetDriver. 
It just looks like this currently:
Screenshot 2023-05-02 at 16 35 08

This PR adds a dedicated `example/exmple.md` file which is used for that page.
It adds a very basic example for how to get started with your first driver.

## Type of Change

- [ ] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [ ] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [ ] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [x] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

## Example

This is how the new example page will look like:

https://user-images.githubusercontent.com/59053438/235699326-a1a61501-f22c-4199-9c48-52865a23cb34.mp4

### CODE REVIEW

File: widget_driver/example/lib/pub_dev_example_code/my_first_drivable_widget.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          FloatingActionButton(
+            onPressed: driver.increaseCounterAction,
+            child: const Icon(Icons.add),
+          ),
+          const SizedBox(width: 10),
+          FloatingActionButton(
+            onPressed: driver.resetCounterAction,
+            child: const Icon(Icons.restore),
```
Comment (JanMichaelPeter) [inline widget_driver/example/lib/pub_dev_example_code/my_first_drivable_widget.dart:23]: You could also put this into the driver

File: widget_driver/example/example.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  @TestDriverDefaultValue()
+  void resetCounterAction() {
+    _count = 0;
+    notifyWidget();
+  }
+}
+```
+
+### 2. Create your Drivable Widget
```
Comment (JanMichaelPeter) [inline widget_driver/example/example.md:63]:
```suggestion
### 3. Run the Code Generator
```

File: widget_driver/example/example.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+Run the WidgetDriver code generator.
+At the root of your flutter project, run this command:
+
+```bash
+flutter pub run build_runner build --delete-conflicting-outputs
+```
+
+### 3. Create your Drivable Widget
```
Comment (JanMichaelPeter) [inline widget_driver/example/example.md:72]:
```suggestion
### 4. Create your Drivable Widget
```

File: widget_driver/example/lib/pub_dev_example_code/my_first_drivable_widget.dart (hunk 1 above)
Comment (robert-northmind) [reply widget_driver/example/lib/pub_dev_example_code/my_first_drivable_widget.dart:23]: Done

File: widget_driver/example/example.md (hunk 2 above)
Comment (robert-northmind) [reply widget_driver/example/example.md:63]: Done

File: widget_driver/example/example.md (hunk 3 above)
Comment (robert-northmind) [reply widget_driver/example/example.md:72]: Done

**Review by 4ndrey — APPROVED** [review]

**Review by JanMichaelPeter — APPROVED** [review]

**Pull request merged**